pinot-core/src/main/java/org/apache/pinot/core/util/SortedRangeIntersection.java

```diff
@@ -60,27 +62,28 @@
         }
       }
       // move all pointers forward such that range they point to contain maxHead
-      for (int i = 0; i < sortedRangeSetList.size(); i++) {
-        if (i == maxHeadIndex) {
+      int j = -1;
```

Comment:
Why do we need to change this?

Reply:
checkstyle doesn't allow modify the for loop value inside the loop block